src/components/PopoverWithMeasuredContent.js

```diff
@@ -59,138 +59,115 @@ const defaultProps = {
-        const horizontalShift = computeHorizontalShift(adjustedAnchorPosition.left, this.popoverWidth, this.props.windowWidth);
-        const verticalShift = computeVerticalShift(adjustedAnchorPosition.top, this.popoverHeight, this.props.windowHeight);
-        const shiftedAnchorPosition = {
-            left: adjustedAnchorPosition.left + horizontalShift,
-            top: adjustedAnchorPosition.top + verticalShift,
-        };
-        return this.state.isContentMeasured ? (
-            <Popover
-                // eslint-disable-next-line react/jsx-props-no-spreading
-                {...this.props}
-                anchorPosition={shiftedAnchorPosition}
-            >
-                {this.props.children}
-            </Popover>
-        ) : (
-            /*
-                This is an invisible view used to measure the size of the popover,
-                before it ever needs to be displayed.
-                We do this because we need to know its dimensions in order to correctly animate the popover,
-                but we can't measure its dimensions without first rendering it.
-            */
-            <View
-                style={styles.invisible}
-                onLayout={this.measurePopover}
-            >
-                {this.props.children}
-            </View>
-        );
-    }
+    };
+
+    const adjustedAnchorPosition = calculateAdjustedAnchorPosition();
+    const horizontalShift = computeHorizontalShift(adjustedAnchorPosition.left, popoverWidth, windowWidth);
+    const verticalShift = computeVerticalShift(adjustedAnchorPosition.top, popoverHeight, windowHeight);
+    const shiftedAnchorPosition = {
+        left: adjustedAnchorPosition.left + horizontalShift,
+        top: adjustedAnchorPosition.top + verticalShift,
+    };
+    return isContentMeasured ? (
+        <Popover
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...props}
+            anchorPosition={shiftedAnchorPosition}
+        >
+            {props.children}
+        </Popover>
+    ) : (
+        /*
+            This is an invisible view used to measure the size of the popover,
+            before it ever needs to be displayed.
+            We do this because we need to know its dimensions in order to correctly animate the popover,
+            but we can't measure its dimensions without first rendering it.
+        */
+        <View
+            style={styles.invisible}
+            onLayout={measurePopover}
+        >
+            {props.children}
+        </View>
+    );
 }
 
 PopoverWithMeasuredContent.propTypes = propTypes;
 PopoverWithMeasuredContent.defaultProps = defaultProps;
+PopoverWithMeasuredContent.displayName = 'PopoverWithMeasuredContent';
 
-export default withWindowDimensions(PopoverWithMeasuredContent);
+export default React.memo(PopoverWithMeasuredContent, (prevProps, nextProps) => prevProps.isVisible && !_.isEqual(prevProps, nextProps));
```

Comment:
This logic is not correct. Previously this was used with the window dimensions props.

Reply:
@s77rt This is not a replace. We change withWindowDimensions to useWindowDimensions hook so hoc is not needed anymore. And we had a `static getDerivedStateFromProps` here https://github.com/Expensify/App/blob/e2e4057ce73b1d0b3762315aed3a1896d48c018a/src/components/PopoverWithMeasuredContent.js#L86-L95 which is replaced with React.memo() with a props comparing function

Reply:
@s77rt This is not a replace. We change withWindowDimensions to useWindowDimensions hook so hoc is not needed anymore. And we had a `shouldComponentUpdate` here https://github.com/Expensify/App/blob/e2e4057ce73b1d0b3762315aed3a1896d48c018a/src/components/PopoverWithMeasuredContent.js#L97-L105 which is replaced with `React.memo` with props comparing function.

Reply:
Checked again and I think we just don't need this after window.. props are removed.

Reply:
What I meant is the comparison logic is not correct. `prevProps.isVisible && !_.isEqual(prevProps, nextProps)` this makes no sense and it seems to be the cause of a regression, the floating action button is broken.

Reply:
I think I'd better return windowWidth/Height props to handle this update logic - if isVisible update only on dimensions change.

Reply:
Let's use the hook and for the `React.memo()` we can omit the function parameters. The "do not update if only window dimensions change" is useless, the component will pretty much already re-render because we are passing a lot of changing data (e.g. anchorPosition is always a new object).

Reply:
No, this leads to a regression with window change. I'll fix the logic

Reply:
Interesting bug. We should fix that bug instead of hiding it. Any ideas yet on the root cause?

Reply:
@s77rt I think the root cause is that shiftedAnchorPosition left/top could be <0 on some updates. You could see it if you add console.log right before return and leave React.memo without props comparing function. So it seems to be not hiding the bug but prevent unnecessary updates.

Reply:
@alexxxwork This still does not explain the root cause. Something is not right, the whole content changes.